### DISCUSSION TITLE
Can 'transform' pull historic state of another entity with 'hass.states'

### DISCUSSION THREAD

**Discussion by paulius2k** [RomRider/apexcharts-card#252] [Q&A] [answered]
Hello,
does anyone know how in the `transform` option I could pull the _historic state_ of another entity - in my case `sensor.thermo_bedroom_temperature` - which was prevailing _at the historic point_ of the chart and not the _current_ state:
```
  - entity: climate.thermo_bedroom
    attribute: hvac_action
    transform: |-
      let tem = ""
      tem = hass.states['sensor.thermo_bedroom_temperature'].state
      if (x === 'heating')
        return tem;
      else
        return 0;
```

What I am trying to do is to imitate the HA graph of thermostat values with `apexcharts-card`, but I cannot manage to get rid of those 'sticking' parts in my apexchart.
I.e.: the idea is to pull the historic value of `hass.states['sensor.thermo_bedroom_temperature'].state` and draw it in another colour at those times when the thermostat `hvac_state` was `heating` - this would 'fill' the area only up to the blue line. Now it just pulls the last value of - "18":

<img src="https://user-images.githubusercontent.com/24705262/144902690-e7493d0e-ab93-44a9-a64e-fcbaf05b39da.png" width=40%>

Thank you.

P.S.: or maybe there's another way how to do it?

**Comment by RomRider** [comment, marked as answer]
That's not possible, but you can cheat by creating a template sensor which follows the temperature line when on and 0 when off.

**Reply by paulius2k** [reply]
True :)  it's a cheat that would work.
